docs/errors/usage_errors.md

```diff
@@ -1238,3 +1237,54 @@ except PydanticUserError as exc_info:
 
 [related specification section]: https://typing.readthedocs.io/en/latest/spec/callables.html#unpack-for-keyword-arguments
 [PEP 692]: https://peps.python.org/pep-0692/
+
+## Invalid `Self` type {#invalid-self-type}
+
+Currently, `Self` can only be used to annotate a field of a class (specifically, subclasses of `BaseModel`, `NamedTuple`, `TypedDict`, or classes decorated by `@dataclass`). Attempting to use `Self` in any other ways will raise this error.
+
+```py
+from typing_extensions import Self
+
+from pydantic import PydanticUserError, validate_call
+
+try:
+
+    @validate_call
+    def func(self: Self):
+        pass
+
+except PydanticUserError as exc_info:
+    assert exc_info.code == 'invalid-self-type'
+```
+
+The following examples of `validate_call` and `TypeAdapter` will also raise this error, even though they are correct from a type-checking perspective. This may be supported in the future.
```

Comment:
```suggestion
The following example of [`validate_call()`][pydantic.validate_call] will also raise this error, even though it is correct from a type-checking perspective. This may be supported in the future.
```